x-pack/metricbeat/module/azure/billing/client.go

```diff
@@ -80,20 +80,15 @@ func (client *Client) GetMetrics(timeOpts TimeIntervalOptions) (Usage, error) {
 		filter,
 		"",  // skipToken, used for paging, not required on the first call.
 		nil, // result page size, defaults to ?
-		consumption.MetrictypeActualCostMetricType,
+		armconsumption.MetrictypeActualCostMetricType,
 		timeOpts.usageStart.Format("2006-01-02"), // startDate
 		timeOpts.usageEnd.Format("2006-01-02"),   // endDate
 	)
 	if err != nil {
 		return usage, fmt.Errorf("retrieving usage details failed in client: %w", err)
 	}
 
-	for paginator.NotDone() {
-		usage.UsageDetails = append(usage.UsageDetails, paginator.Values()...)
-		if err := paginator.NextWithContext(context.Background()); err != nil {
-			return usage, fmt.Errorf("retrieving usage details failed in client: %w", err)
-		}
-	}
+	usage.UsageDetails = append(usage.UsageDetails, paginator.Value...)
```

Comment:
Since we are not paginating anymore in this function, maybe we can drop `paginator` and use something different, like `result` (it's just an example).